src/mps/projmposum.jl

```diff
@@ -128,10 +98,68 @@ ProjMPOSum `P`, and `ortho` is a String which can take
 the values `"left"` or `"right"` depending on the
 sweeping direction of the DMRG calculation.
 """
-function noiseterm(P::ProjMPOSum, phi::ITensor, dir::String)
-  nt = noiseterm(P.pm[1], phi, dir)
-  for n in 2:length(P.pm)
-    nt += noiseterm(P.pm[n], phi, dir)
+function noiseterm(P::AbstractSum, phi::ITensor, dir::String)
+  nt = noiseterm(P.terms[1], phi, dir)
+  for n in 2:length(P.terms)
+    nt += noiseterm(P.terms[n], phi, dir)
   end
   return nt
 end
+
+#
+# Definition of concrete, generic ProjSum type
+#
+
+mutable struct ProjSum{T} <: AbstractSum
+  terms::Vector{T}
+end
+
+ProjSum{T}(mpos::Vector{MPO}) where {T} = ProjSum([T(M) for M in mpos])
+
+ProjSum{T}(Ms::MPO...) where {T} = ProjSum{T}([Ms...])
+
+"""
+A ProjMPOSum computes and stores the projection of an
+implied sum of MPOs into a basis defined by an MPS,
+leaving a certain number of site indices of each MPO
+unprojected. Which sites are unprojected can be shifted
+by calling the `position!` method. The MPOs used as
+input to a ProjMPOSum are *not* added together beforehand;
+instead when the `product` method of a ProjMPOSum is invoked,
+each projected MPO in the set of MPOs is multiplied by
+the input tensor one-by-one in an efficient way.
+
+Drawing of the network represented by a ProjMPOSum
+`P([H1,H2,...])`, showing the case of `nsite(P)==2`
+and `position!(P,psi,4)` for an MPS `psi` (note the
+sum Σⱼ on the left):
+
+```
+     o--o--o-      -o--o--o--o--o--o <psi|
+     |  |  |  |  |  |  |  |  |  |  |
+ Σⱼ  o--o--o--o--o--o--o--o--o--o--o Hⱼ
+     |  |  |  |  |  |  |  |  |  |  |
+     o--o--o-      -o--o--o--o--o--o |psi>
+```
+"""
+const ProjMPOSum = ProjSum{ProjMPO}
+
+"""
+A `DiskProjMPOSum` functions the same as a `ProjMPOSum`
+(see the docstring for `ProjMPOSum`) but automatically caches most of
+the tensors it stores onto the hard drive to save RAM memory.
+"""
+const DiskProjMPOSum = ProjSum{DiskProjMPO}
+
+"""
+    disk(ps::ProjMPOSum; kwargs...)
+
+Convert a ProjMPOSum into a DiskProjMPOSum,
+which will automatically start caching most 
+stored tensors onto the hard drive.
+"""
+function disk(ps::ProjMPOSum; kwargs...)
+  return DiskProjMPOSum([disk(pm; kwargs...) for pm in ps.terms])
+end
```

Comment:
I think my previous comment got lost, but I think we should make this a generic `AbstractSum` function:
```julia
function disk(sum::AbstractSum; disk_kwargs...)
  return set_terms(sum, [disk(term; disk_kwargs...) for term in terms(sum)])
end
```
based on the generic `set_terms` function I propose above. That will be helpful for the new `ITensorParallel.ParallelSum` type.